src/main/host/descriptor/mod.rs

```diff
@@ -303,6 +309,20 @@ impl PosixFile {
     }
 }
 
+impl std::fmt::Debug for PosixFile {
```

Comment:
Is this substantially different from what derive(Debug) would get us? I *think* it already stringifies enum values

Reply:
The `derive(Debug)` impl for an enum will also print whatever objects it contains. This means that all file types (ex: `PipeFile`) then also need to impl `Debug`, and if you `println!("{:?}", desc)`, it will print out a huge amount of information. I'm assuming that if someone does a `println!("{:?}", desc)`, then they're probably only concerned about the high-level details like the descriptor/file type, status, and flags, and not all the details about its contents (for example the pipe's internal buffer, event source, event handle, etc).

Reply:
Ah, makes sense